state/txindex/kv/kv.go

```diff
@@ -461,9 +460,6 @@ func (txi *TxIndex) match(
 		defer it.Close()
 
 		for ; it.Valid(); it.Next() {
-			if !isTagKey(it.Key()) {
-				continue
-			}
```

Comment:
Is this necessary? Can't we assume that all keys with the same prefix will have the desired format

Reply:
You should never assume anything for externally sourced data. For example, users trying to upgrade/downgrade and reuse the same database might hit this if the key format changes.